ChronoKeeper/StoryPipeline.cpp

```diff
@@ -1,44 +1,271 @@
 
+#include <deque>
 #include <map>
 #include <mutex>
 
-#include "KeeperDataStore.h"
+
+#include "StoryChunk.h"
+#include "StoryPipeline.h"
+#include "StoryIngestionHandle.h"
+
+#define TRACE_CHUNKING
+
+namespace chl = chronolog;
 
 ////////////////////////
 
-chronolog::StoryPipeline::StoryPipeline( chronolog::StoryId const& story_id, std::mutex & ingestion_mutex, std::mutex & sequencing_mutex)
+chronolog::StoryPipeline::StoryPipeline( std::string const& chronicle_name, std::string const& story_name,  chronolog::StoryId const& story_id
+		, uint64_t story_start_time, uint16_t chunk_granularity, uint16_t archive_granularity, uint16_t acceptance_window)
 	: storyId(story_id)
-	, ingestionMutex(ingestion_mutex)
-	, sequencingMutex(sequencing_mutex)  
+	, chronicleName(chronicle_name)
+	, storyName(story_name)
+	, timelineStart(story_start_time)   
+	, timelineEnd(story_start_time)
+	, chunkGranularity(chunk_granularity)
+	, archiveGranularity(archive_granularity)  
+	, acceptanceWindow(acceptance_window)  
+	, activeIngestionHandle(nullptr)
 
 {
+	activeIngestionHandle = new chl::StoryIngestionHandle( ingestionMutex, &eventQueue1, &eventQueue2);
+
+	//pre-initialize the pipeline map with the StoryChunks of chunkGranulary 
+	// with the total time window of max(acceptance_window, chunk_granularity*2) 
+
+    	auto story_start_point = std::chrono::time_point<std::chrono::system_clock,std::chrono::nanoseconds>{}
```

Comment:
system_clock is not guaranteed to be monotonic. I think we should use steady_clock which is monotonic.

Reply:
Please, note that I chrono::system_clock here is only used for logging, not as a timestamp of any sort. the timelineStart is the value that comes from the Vizor in the StoryAcquisition message